middleware/instrument.go

```diff
@@ -68,9 +68,20 @@ func (i Instrument) Wrap(next http.Handler) http.Handler {
 			next.ServeHTTP(ww, r)
 		})
 
-		i.Duration.WithLabelValues(r.Method, route, strconv.Itoa(respMetrics.Code), isWS).Observe(respMetrics.Duration.Seconds())
 		i.RequestBodySize.WithLabelValues(r.Method, route).Observe(float64(rBody.read))
 		i.ResponseBodySize.WithLabelValues(r.Method, route).Observe(float64(respMetrics.Written))
+
+		if traceID, ok := ExtractSampledTraceID(r.Context()); ok {
+			i.Duration.WithLabelValues(r.Method, route, strconv.Itoa(respMetrics.Code), isWS).
```

Comment:
suggest a different structure to avoid repeating parts:
```
observer := i.Duration.WithLabelValues(r.Method, route, strconv.Itoa(respMetrics.Code), isWS)
if ... {
    observer.(prometheus.ExemplarObserver).ObserveWithExemplar()
} else {
    observer.Observe()
}
```

Reply:
Done. Thanks.